provider/datagen/src/transform/segmenter/mod.rs

```diff
@@ -267,6 +119,155 @@ impl crate::DatagenProvider {
         let data = sets::load_extended_pictographic(self).expect("The data should be valid!");
         let extended_pictographic = data.as_borrowed();
 
+        fn set_break_state(
+            break_state_table: &mut [i8],
+            property_length: usize,
+            left_index: usize,
+            right_index: usize,
+            break_state: i8,
+        ) {
+            let index = left_index * property_length + right_index;
+            if break_state_table[index] == UNKNOWN_RULE
+                || break_state_table[index] == NOT_MATCH_RULE
+            {
+                break_state_table[index] = break_state;
+            }
+        }
+
+        fn get_index_from_name(properties_names: &[String], s: &str) -> Option<usize> {
+            properties_names.iter().position(|n| n.eq(s))
+        }
+
+        fn get_word_segmenter_value_from_name(name: &str) -> WordBreak {
+            match name {
+                "ALetter" => WordBreak::ALetter,
+                "CR" => WordBreak::CR,
+                "Double_Quote" => WordBreak::DoubleQuote,
+                "Extend" => WordBreak::Extend,
+                "ExtendNumLet" => WordBreak::ExtendNumLet,
+                "Format" => WordBreak::Format,
+                "Katakana" => WordBreak::Katakana,
+                "Hebrew_Letter" => WordBreak::HebrewLetter,
+                "LF" => WordBreak::LF,
+                "MidLetter" => WordBreak::MidLetter,
+                "MidNum" => WordBreak::MidNum,
+                "MidNumLet" => WordBreak::MidNumLet,
+                "Newline" => WordBreak::Newline,
+                "Numeric" => WordBreak::Numeric,
+                "Regional_Indicator" => WordBreak::RegionalIndicator,
+                "Single_Quote" => WordBreak::SingleQuote,
+                "WSegSpace" => WordBreak::WSegSpace,
+                "ZWJ" => WordBreak::ZWJ,
+                _ => {
+                    panic!("Invalid property name")
+                }
+            }
+        }
+
+        fn get_grapheme_segmenter_value_from_name(name: &str) -> GraphemeClusterBreak {
```

Comment:
Issue: This is a complicated PR, and it is more difficult to review when you refactor code in a way that is unrelated to the change. It appears that you changed this function from being a module function to being an inline function. It would be a much smaller change if you would add `cfg`s to the module functions if that's all you need to make the code compile without warnings.

I didn't check if you have this in a standalone commit; please use more detailed commit messages if you prefer a commit-by-commit review.

Reply:
This is the most recent commit called `seg cleanup`. I would have preferred to make this a standalone PR, but with the cross-timezone review cycle having chains is a massive pain.

Reply:
OK thanks for the pointer; everything looks mergeable now except for the `seg cleanup` commit which requires more of my time so I will look at in more detail this afternoon.

If your goal is to get things reviewed faster, please don't scope-creep them in the middle of a review cycle.

Reply:
@robertbastian highly recommend just making a separate branch and creating a draft PR with "based on <other PR>, start reviewing at <commit name>"

@sffc I think given that this is a large PR already we should tend towards merging sooner rather than later and deal with the leftover stuff as post-review.

Reply:
> @robertbastian highly recommend just making a separate branch and creating a draft PR with "based on , start reviewing at "

This has not worked well for me in the past, I would get approval it but it has merge conflicts, I lose approval and it takes another couple of days to get approval again.

Reply:
I find rebases to be better than merges for dealing with that. But yeah that can be a pain.

Reply:
My preferred solution is to make a new PR as Draft, and once the parent PR is merged, fix up the child Draft PR and then send it out for review. Avoid PR chains longer than 2 or 3 by working on some other component that is independent; the project is big enough that there are lots of options.